datadog/resource_datadog_monitor.go

```diff
@@ -466,8 +466,8 @@ func resourceDatadogMonitorUpdate(d *schema.ResourceData, meta interface{}) erro
 	datadogClientV1 := providerConf.DatadogClientV1
 	authV1 := providerConf.AuthV1
 
-	m := buildMonitorStruct(d)
-
+	monitor := buildMonitorStruct(d)
+	m := datadogV1.MonitorUpdateRequest(*monitor)
```

Comment:
let's not try to convert type here, and do like for SLO with `buildMonitorStruct` return two types

Reply:
Oups good catch!! Thx